src/main/java/org/mitre/synthea/engine/State.java

```diff
@@ -207,6 +207,10 @@ public boolean process(Person person, long time) {
       // e.g. "submodule": "medications/otc_antihistamine"
       List<State> moduleHistory = person.history;
       Module submod = Module.getModuleByPath(submodule);
+      HealthRecord.Encounter encounter = person.getCurrentEncounter(module);
```

Comment:
I think `ConditionOnset` is different because we say that conditions can be present without being diagnosed, so I'm not sure it would be correct to change it to be consistent with the others, which should just use the current encounter.

Reply:
The ConditionOnset isn't put into the record until it's diagnosed though, and that only happens at the current encounter. The only difference is whether it's using `person.getCurrentEncounter(module)` or `record.currentEncounter(time)`. The reason this "encounter-by-module" concept exists is that nothing in the engine prevents a person from being in multiple encounters at once, it's just a very low probability of happening, and even if it does there's a low probability of conflicts. So getting the encounter by time means that there's a small chance of unexpected results.

Reply:
> and that only happens at the current encounter

Or the "target_encounter" which might not be the same thing.

Reply:
The diagnosis happens when target_encounter == current encounter.